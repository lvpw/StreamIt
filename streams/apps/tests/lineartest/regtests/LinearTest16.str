/** Test out combining splitjoins. **/
void->void pipeline LinearTest16 {
  add FloatSource();
  add LinearSJ();
  add LinearSJ2();
  add FloatSink();
}

void->float stateful filter FloatSource {
  float x;
  init {
    x = 0;
  }
  work push 1 {
    push(x);
    x = x + 1;
  }
}

float->void filter FloatSink {
  work pop 1 {
    println(pop());
  }
}

/** A linear split join to test combination with different round robin weightings. **/
float->float splitjoin LinearSJ {
  split roundrobin(2,3);
  add FloatFilter1();
  add FloatFilter2();
  join roundrobin(2,3);
}

float->float splitjoin LinearSJ2 {
  split roundrobin(4,6);
  add FloatFilter1();
  add FloatFilter2();
  join roundrobin(4,6);
}


/** filter that computes [[1 2][3 4]] **/
float->float filter FloatFilter1 {
  work pop 2 push 2 {
    float t1 = pop();
    float t2 = pop();
    push(t2*2 + t1*4);
    push(t2*1 + t1*3);
  }
}
/** filter that computes [[5 6 7][8 9 10][11 12 13]] **/
float->float filter FloatFilter2 {
  work pop 3 push 3 {
    float t1 = pop();
    float t2 = pop();
    float t3 = pop();
    push(13*t1 + 10*t2 + 7*t3);
    push(12*t1  + 9*t2 + 6*t3);
    push(11*t1  + 8*t2 + 5*t3);
  }
}
