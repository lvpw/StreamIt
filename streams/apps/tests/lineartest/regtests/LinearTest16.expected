FloatFilter1_1.matrix
[[1.0+0.0i 2.0+0.0i]
 [3.0+0.0i 4.0+0.0i]]
FloatFilter1_1.vector
[[0.0+0.0i 0.0+0.0i]]
FloatFilter1_2.matrix
[[1.0+0.0i 2.0+0.0i]
 [3.0+0.0i 4.0+0.0i]]
FloatFilter1_2.vector
[[0.0+0.0i 0.0+0.0i]]
FloatFilter2_1.matrix
[[5.0+0.0i 6.0+0.0i 7.0+0.0i]
 [8.0+0.0i 9.0+0.0i 10.0+0.0i]
 [11.0+0.0i 12.0+0.0i 13.0+0.0i]]
FloatFilter2_1.vector
[[0.0+0.0i 0.0+0.0i 0.0+0.0i]]
FloatFilter2_2.matrix
[[5.0+0.0i 6.0+0.0i 7.0+0.0i]
 [8.0+0.0i 9.0+0.0i 10.0+0.0i]
 [11.0+0.0i 12.0+0.0i 13.0+0.0i]]
FloatFilter2_2.vector
[[0.0+0.0i 0.0+0.0i 0.0+0.0i]]
LinearSJ2_1.matrix
[[5.0+0.0i 6.0+0.0i 7.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [8.0+0.0i 9.0+0.0i 10.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [11.0+0.0i 12.0+0.0i 13.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 5.0+0.0i 6.0+0.0i 7.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 8.0+0.0i 9.0+0.0i 10.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 11.0+0.0i 12.0+0.0i 13.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 1.0+0.0i 2.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 3.0+0.0i 4.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 1.0+0.0i 2.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 3.0+0.0i 4.0+0.0i]]
LinearSJ2_1.vector
[[0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]]
LinearSJ_1.matrix
[[5.0+0.0i 6.0+0.0i 7.0+0.0i 0.0+0.0i 0.0+0.0i]
 [8.0+0.0i 9.0+0.0i 10.0+0.0i 0.0+0.0i 0.0+0.0i]
 [11.0+0.0i 12.0+0.0i 13.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 1.0+0.0i 2.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 3.0+0.0i 4.0+0.0i]]
LinearSJ_1.vector
[[0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]]
LinearWrapper_1.matrix
[[150.0+0.0i 168.0+0.0i 186.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [222.0+0.0i 249.0+0.0i 276.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [294.0+0.0i 330.0+0.0i 366.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 21.0+0.0i 24.0+0.0i 27.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 47.0+0.0i 54.0+0.0i 61.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 55.0+0.0i 60.0+0.0i 65.0+0.0i 27.0+0.0i 40.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 88.0+0.0i 96.0+0.0i 104.0+0.0i 39.0+0.0i 58.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 121.0+0.0i 132.0+0.0i 143.0+0.0i 51.0+0.0i 76.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 7.0+0.0i 10.0+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 15.0+0.0i 22.0+0.0i]]
LinearWrapper_1.vector
[[0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]]
